diesel/src/sqlite/connection/raw.rs

```diff
@@ -379,6 +400,40 @@ unsafe fn null_aggregate_context_error(ctx: *mut ffi::sqlite3_context) {
     );
 }
 
+#[allow(warnings)]
+extern "C" fn run_collation_function<F>(
+    user_ptr: *mut libc::c_void,
+    len_rhs: libc::c_int,
+    rhs_ptr: *const libc::c_void,
+    len_lhs: libc::c_int,
+    lhs_ptr: *const libc::c_void,
+) -> libc::c_int
+where
+    F: FnMut(&str, &str) -> std::cmp::Ordering + Send + 'static,
+{
+    unsafe {
+        let user_ptr = user_ptr as *mut F;
+        let f = match user_ptr.as_mut() {
+            Some(f) => f,
+            None => {
+                //FIXME
```

Comment:
I'm not really sure what to do here if sqlite passes us null pointers. Can it just be assumed that this will not happen (i.e. `unwrap()` here)?

Reply:
It seems to me `unreachable!` in case of `None` could be appropriate. 
https://github.com/diesel-rs/diesel/blob/44d0096ae99d135a644a2cb26da11b6a5135de90/diesel/src/sqlite/connection/raw.rs#L307-L309

Reply:
It wouldn't be, since "A panic! across an FFI boundary is undefined behavior." (https://doc.rust-lang.org/nomicon/ffi.html#ffi-and-panics).

Reply:
According to https://www.sqlite.org/c3ref/create_collation.html we can be sure that we get our pouter to the callback here. Following that reasoning I would use `unreachable!` or something similar + a comment explaining why this cannot happen to indicate that this is guaranteed by other things.

Reply:
Then I'd either go to "full undefined behavior" territory and use `std::hint::unreachable_unchecked()`, or stay on the safe route and call `std::process::abort()` (with an `eprintln!()` in front).
IMO `unreachable!()` gives the impression that we can just panic and it's fine. Which it may eventually be (rust-lang/rust#74990), but at the moment it is not.

Reply:
Using `eprintln!` + `abort()` sounds like the better solution for me. Can you have a quick lock at the other callbacks for normal and aggregate functions and change that there as well?

Reply:
Yes, will do!